### DISCUSSION TITLE
关于为多个文件夹设置密码

### DISCUSSION THREAD

**Discussion by Qubik65536** [spencerwooo/onedrive-vercel-index#170] [Q&A] [answered]
您好，

有个问题，假如我在 `config/site.json` 中设置：

```
"protectedRoutes": [
  "/Private folder"
],
```

或者增加 `/` 或者通配符 `*`，我是否能够为 `Private folder` 下每个文件夹单独添加 `.password` 文件设置独立的密码？

感谢解答

**Comment by spencerwooo** [comment, marked as answer]
`protectedRoutes` 在项目中的 [`utils/tools.ts`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/utils/tools.ts#L99-L125) 进行匹配的，因此：

1. 通配符 `*` 一定是不可以的，因为是根据当前路径来「匹配是否为加密文件夹」并「找到当前加密文件夹的 `.password` 路径」的，而不是通配符则匹配全部路径；
2. 直接设置根目录 `/` 也无法完成你想要的效果，因为是根据 match 到的路径来确定当前文件夹所在的 `.password` 绝对路径，如果设置为 `/` 就会默认找到 `~/.password` 这一个文件而不是你所要的「每个文件夹的单独 `.password` 文件」来「独立设置密码」。

**Reply by Qubik65536** [reply]
那请问这个功能有望在以后的版本中实现吗？建立这个站点我主要的目的是分享一些东西，而我们有很多个上传资源的人，但是大部分人都不会使用GitHub和这类工具

**Reply by spencerwooo** [reply]
不会，设置 `protectedRoutes` 的初衷在于让访问某个目录的时候方便找到目录对应的 `.password` 文件 — 这个文件可能在当前文件夹下，也可能在上级文件夹里面。如果像你这样配置，那么当前逻辑下肯定是无法找到对应的 `.password` 文件，也就无法判断当前目录是否是 protected 的文件夹了。当然，这就是 serverless 的弊端 — 不过如果你有更好的解决方案，当然我也来之不拒。

**Reply by Qubik65536** [reply]
明白了，感谢解答

------

由于对serverless开发不熟悉，我仅能提出一个甚至可能不可行的想法：我们是否可以设立另一个独立于 `protectedRoutes` 的参数（或者对该参数直接进行改动），而我们会把这个新参数里的路径下的一级路径认定为 protected 文件夹，并从这些文件夹里找到 `.password`。如果我说的有问题，也请谅解
